sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceUtils.java

```diff
@@ -320,6 +320,7 @@ public static Resource getExpectedRanges(
                 .getLabelsBuilder()
                 .addLabels(getVIPLabel(vipName, vipPort));
 
+        builder.setVisibility(DiscoveryInfo.Visibility.CLUSTER);
```

Comment:
NamedVIPEvaluationStage.java:L95 
Task visibility should be enough, this is to ensure discovery visibility is always CLUSTER.